### ISSUE TITLE
Dependency on glibc causes credhub-cli to not work in alpine

### ISSUE THREAD

**Issue by akshaymankar** [cloudfoundry-incubator/credhub-cli#55]
**What version of the credhub server you are using?**
Doesn't matter

**What version of the credhub cli you are using?**
2.0.0

**If you were attempting to accomplish a task, what was it you were attempting to do?**
Doing anything with cli.

**What did you expect to happen?**
I expected it to work.

**What was the actual behavior?**
```
$ docker run -it --rm bash bash
bash-4.4# apk add -u wget
bash-4.4# wget https://github.com/cloudfoundry-incubator/credhub-cli/releases/download/2.0.0/credhub-linux-2.0.0.tgz
bash-4.4# tar xvf credhub-linux-2.0.0.tgz
./
./credhub
bash-4.4# ./credhub
bash: ./credhub: No such file or directory
bash-4.4# ldd ./credhub
    /lib64/ld-linux-x86-64.so.2 (0x7ff878a46000)
    libpthread.so.0 => /lib64/ld-linux-x86-64.so.2 (0x7ff878a46000)
    libc.so.6 => /lib64/ld-linux-x86-64.so.2 (0x7ff878a46000)
```

This forces me to use heavier images or somehow install glibc on alpine. It would be nice to not depend on glibc.

* [x] I have included a log output
* [x] My log includes an error message
* [x] I have included steps for reproduction

**Comment by jtarchie** [comment]
Does this just require compiling the linux binary with `CGO_ENABLED=0` to be set?

**Comment by akshaymankar** [comment]
@jtarchie I just tried it and I can confirm it works. So, maybe it is just a small change in credhub pipelines :)

**Comment by aryanet** [comment]
Bumped into the same issue.

**Comment by aryanet** [comment]
@athornton2012 this blocks us from using credhub cli in our CD pipelines that use Alpine Linux. Any idea of a workaround?

**Comment by jtarchie** [comment]
This may not be as small a change as a originally thought. Depending on the dependencies of credhub-cli, they may need CGO.

@aryanet, I'd recommend from using `alpine` based images for CD pipelines. There is no guarantee for the origin of the update that Alpine provides. It might be a smaller image, but a community keeps Alpine up to date. These may or may not include timely updates. A ubuntu image will guarantee CVE updates in timely manner, and is officially supported from docker and Canonical.

I've slowly become a proponent of using Ubuntu image for these reasons. Especially CD/CI pipelines. They should be as secure (if not more) than local development.

As for the argument of _heavy images_. Can I ask why they are an issue? Most CI/CD systems have really good internet connections and caching layers, so having a heavy image isn't expensive.

**Comment by arthfl** [comment]
Just ran into the same issue

@jtarchie Regarding image sizes:
Why should one prefer bigger images over smaller ones?
It's not only about connection speed (although that is also a very valid factor), memory to host images costs money.

**Comment by jtarchie** [comment]
@arthfl, thanks for the reply.

My initial point about stable and secure images is an important. Do you know where your linux distro is coming from and have a confident chain of authenticity for patches being applied to it?

When it comes to size, [GCR](https://cloud.google.com/container-registry/pricing) is 2 cents USD per GB and [AWS ECR](https://aws.amazon.com/ecr/pricing/) is 10 cents USD per GB. [Azure](https://azure.microsoft.com/en-us/pricing/details/container-registry/) has a daily pricing tier. Public images on docker hub are free.

The cost of a secure and stable image (Ubuntu) is more important than the cost of a hosted image size.

**Comment by arthfl** [comment]
@jtarchie 
Public images are definitly not always an option. Since they might just trigger support-cases if customers stumble upon them. And yes, space to safe images isn't the most expensive thing in public clouds, but things accumulate :-)
And some of us are running internal registries too, which need disk space, which costs money.

Also, stating that Alpine isn't at least equally ecure as Ubuntu is quite bold.

Anyways, we're quite off-topic here ;-)

**Comment by martys34** [comment]
@arthfl We will pull our PM's (@jparnala @LizClaireMorris) into this thread so that they can prioritize the issue as necessary. However, we have some high priority work in flight at the moment so I'm not sure if or when this will be prioritized.

**Comment by ankeesler** [comment]
We are working on a chore that we think will address this issue: https://www.pivotaltracker.com/story/show/171286204.

**Comment by tomkennedy513** [comment]
This is being prioritized now

**Comment by ankeesler** [comment]
We believe this issue has been resolved as of the 2.7.1 beta build. Check it out here: https://storage.googleapis.com/credhub-cli-tarballs/credhub-linux-2.7.1-beta.1.tgz. We should have this out in a GA release within a week or two. Let us know if this does not fix the issue.
